From 2867804041ccf92a5c7700eaa946d743565653ec Mon Sep 17 00:00:00 2001
From: Richard Feliciano <rfeliciano@BeaconEmbedded.com>
Date: Fri, 13 Jan 2023 15:20:59 -0600
Subject: [PATCH] Change dram init memory code to handle 8GB config for RZG2H

Modified the dram init code so it could access 8GB memory on RZG2
Renesas H platforms

RSOM-433] RZG2H 8GB RAM

Signed-off-by:Richard Feliciano  <rfeliciano@beaconembedded.com>

diff --git a/drivers/renesas/common/ddr/ddr_b/boot_init_dram_config.c b/drivers/renesas/common/ddr/ddr_b/boot_init_dram_config.c
index 04fe64f59..755f91109 100644
--- a/drivers/renesas/common/ddr/ddr_b/boot_init_dram_config.c
+++ b/drivers/renesas/common/ddr/ddr_b/boot_init_dram_config.c
@@ -376,74 +376,74 @@ static const struct _boardcnf boardcnfs[BOARDNUM] = {
 	},
 /* boardcnf[6] Beacon Embedded G2H/SOC */
 {
-	0x05,
-	0x01,
-	0x0320,
-	0,
-	0x0300,
-	0x00a0,
-	{
-	{
-	   {0x04, 0xff},
-	   0x00345201,
-	   0x3201,
-	   {0x23617540, 0x54067312, 0x43251607, 0x71605243},
-	   {0x08, 0x08, 0x08, 0x08},
-	   WDQLVL_PAT,
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0},
-	   {0, 0, 0, 0},
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0},
-	   {0, 0, 0, 0},
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0}
-	},
-	{
-	   {0x04, 0xff},
-	    0x00302154,
-	    0x2310,
-	   {0x26015734, 0x04531276, 0x43251607, 0x10647532},
-	   {0x08, 0x08, 0x08, 0x08},
-	   WDQLVL_PAT,
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0},
-	   {0, 0, 0, 0 },
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0},
-	   {0, 0, 0, 0},
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0}
-	   },
-	{
-	   {0x04, 0xff},
-	    0x00302154,
-	    0x2310,
-	   {0x26015734, 0x04531276, 0x43251607, 0x10647532},
-	   {0x08, 0x08, 0x08, 0x08},
-	   WDQLVL_PAT,
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0},
-	   {0, 0, 0, 0},
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0},
-	   {0, 0, 0, 0},
-	   {0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0,
-	    0, 0, 0, 0, 0, 0, 0, 0}
-	},
-	{
+	0x05,		/* phyvalid */
+	0x01,		/* dbi_en */
+	0x0320,		/* cacs_dly */
+	0,		/* cacs_dly_adj */
+	0x0300,		/* dqdm_dly_w */
+	0x00a0,		/* dqdm_dly_r */
+	{
+/*ch[0]*/	{
+/*ddr_density[]*/	{0x04, 0x04},
+/*ca_swap*/		0x00345201,
+/*dqs_swap[]*/	   	0x3201,
+/*dq_swap[]*/	   	{0x23617540, 0x54067312, 0x43251607, 0x71605243},
+/*dm_swap[]*/	   	{0x08, 0x08, 0x08, 0x08},
+/*wdqlvl_patt[]*/	WDQLVL_PAT,
+/*cacs_adj*/		{0, 0, 0, 0, 0, 0, 0, 0,
+	    		0, 0},
+/*dm_adj_w*/		{0, 0, 0, 0},
+/*dqdm_adj_w*/		{0, 0, 0, 0, 0, 0, 0, 0,
+	    		0, 0, 0, 0, 0, 0, 0, 0,
+	    		0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0},
+/*dm_adj_r*/		{0, 0, 0, 0},
+/*dqdm_adj_r*/		{0, 0, 0, 0, 0, 0, 0, 0,
+	    		0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0}
+		},
+/*ch[1]*/	{
+/*ddr_density[]*/	{0x04, 0x04},
+/*ca_swap*/		0x00302154,
+/*dqs_swap[]*/		0x2310,
+/*dq_swap[]*/	   	{0x26015734, 0x04531276, 0x43251607, 0x10647532},
+/*dm_swap[]*/	   	{0x08, 0x08, 0x08, 0x08},
+/*wdqlvl_patt[]*/	WDQLVL_PAT,
+/*cacs_adj*/		{0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0},
+/*dm_adj_w*/		{0, 0, 0, 0 },
+/*dqdm_adj_w*/		{0, 0, 0, 0, 0, 0, 0, 0,
+	    		0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0},
+/*dm_adj_r*/		{0, 0, 0, 0},
+/*dqdm_adj_r*/		{0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0}
+	   	},
+/*ch[2]*/	{
+/*ddr_density[]*/	{0x04, 0x04},
+/*ca_swap*/		0x00302154,
+/*dqs_swap[]*/		0x2310,
+/*dq_swap[]*/		{0x26015734, 0x04531276, 0x43251607, 0x10647532},
+/*dm_swap[]*/		{0x08, 0x08, 0x08, 0x08},
+/*wdqlvl_patt[]*/	WDQLVL_PAT,
+/*cacs_adj*/		{0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0},
+/*dm_adj_w*/		{0, 0, 0, 0},
+/*dqdm_adj_w*/		{0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0},
+/*dm_adj_r*/		{0, 0, 0, 0},
+/*dqdm_adj_r*/		{0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0}
+		},
+/*dummy*/	{
 	   {0xff, 0xff},
 	    0,
 	    0,
-- 
2.34.1

